### DISCUSSION TITLE
STK audio effects implementation

### DISCUSSION THREAD

**Discussion by eccessivo75** [pschatzmann/arduino-audio-tools#1026] [Q&A] [answered]
Hello,
As anyone been able to use an stk audio effect?
I wanted to try the audio effect, and they worked well , all but the STK.
First the header stkall.h was missing , so i added this library to make it work: https://github.com/pschatzmann/Arduino-STK
After this trying to compile a sketch with a  STKlentpitshift implemantation i get this error:
Cannot declare variable 'stkps' to be of abstract type 'audio_tools::STKLentPitShift . This is related to this line ot my code:
STKLentPitShift stkps;
No luck , with the same error for the STKpitchshift.
I am actually able to successfully compile the STKNreverb... but it crashes on my LyraT test board (probably a memory issue) .
Let me know if i am doing wrong.

**Comment by pschatzmann** [comment, marked as answer]
Not sure how this happend:
I committed some corrections to the AudioTools and arduino-stk to make the compile errors go way...

**Reply by eccessivo75** [reply]
Hi,
thankyou for the fast answer. I was still not successful.
I updated the audio tools lib and the arduino stk lib, but i get the same error. I deleted and reinstalled the libs. I was able to see the updated file inside the arduino stk lib. Not the same for the audiotools lib it seems unchanged
let me know

**Comment by pschatzmann** [comment]
Your right: it seems that it has not been pushed to github.
The [latest change](https://github.com/pschatzmann/arduino-audio-tools/commit/e38edd7269786178ba18ac8847b24666f1a0f026) is available now...

**Reply by eccessivo75** [reply]
Hi,  i can confirm that the issue has  gone i am able to compile  the code as expected. The STKlentpitshift is still not working: it compiles but produces cracking sound on my LyraT board. The STK pitch shifter is working fine.
I think it is mostly related to my Hardware . Thank you.
